### PR TITLE
tidb: move loadCommonGlobalVariablesIfNeeded to Execute

### PR DESCRIPTION
**Pull request by tiancaiamao**
build plan depends on time_zone system variable, if a plan is build
before time_zone loaded, the generated plan may be wrong.
So we should not loadCommonGlobalVariablesIfNeeded until ActivePendingTxn.
This change ensure that time_zone is loaded before plan build.

@shenli @coocood

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: session.go [hunk 1]
```diff
@@ -727,6 +727,11 @@ func (s *session) Execute(goCtx goctx.Context, sql string) (recordSets []ast.Rec
 			return nil, errors.Trace(err)
 		}
 	} else {
+		err = s.loadCommonGlobalVariablesIfNeeded()
```
Comment (shenli) [inline session.go]: How about use prepared statement to run SQL?

Comment (tiancaiamao) [reply session.go]: Fixed

**Comment by zimulala** [comment]
LGTM

**Pull request merged**